### PR TITLE
[TS] LPS-176493 Facets show results with Empty Searches with a Blueprint on Page

### PR DESCRIPTION
**Pull request by joshuacords**
https://issues.liferay.com/browse/LPS-176493

Continued from https://github.com/liferay-search/liferay-portal/pull/815

**Author:** @joshuacords
**Reviewers:** @gustavolimav @BryanEngler

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:test:search

**Comment by xbrianlee** [comment]
Most of the integration test failures are related to this bug fix. Every test marked "failed" here is related:
https://testray.liferay.com/home/-/testray/case_results/index?delta=20&p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=2224277620&testrayRunId=2224277663&statuses=3

**Comment by joshuacords** [comment]
[UserIndexerIndexedFieldsByAssociationTest](https://testray.liferay.com/home/-/testray/case_results/2228667902) was actually caused by this. I'll work on the other ones, thanks Brian!

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/builder/AssetCategoriesSearchFacetDisplayContextBuilder.java [hunk 1]
```diff
@@ -377,6 +377,10 @@ protected void setTermDisplayContexts(
 	}
 
 	private List<Tuple> _collectBuckets(Facet facet) {
+		if (facet == null) {
```
Comment (BryanEngler) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/builder/AssetCategoriesSearchFacetDisplayContextBuilder.java:380]: @joshuacords is this check still necessary? seems odd that only category facet would need it?

Comment (joshuacords) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/builder/AssetCategoriesSearchFacetDisplayContextBuilder.java:380]: It's probably not necessary any more, but I think the Category and Vocabulary facet were doing permission checking on the facets and that's why just they needed it. (Folders does a followup search to check).
So I don't think it's necessary, but I don't see why we won't want to handle the null situation - do you think I should remove that and test again? @BryanEngler

**Comment by xbrianlee** [comment]
Whoops, forgot to leave a comment yesterday. The latest `ci:test:search` results look good.

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/builder/AssetCategoriesSearchFacetDisplayContextBuilder.java (hunk 1 above)
Comment (BryanEngler) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/builder/AssetCategoriesSearchFacetDisplayContextBuilder.java:380]: actually it looks like all of the other FacetDisplayContextBuilders are already doing this check, this was the only one that wasnt so should be fine to leave the check there

**Comment by joshuacords** [comment]
ci:forward

**Comment by BryanEngler** [comment]
ci:stop

**Comment by BryanEngler** [comment]
@joshuacords see https://github.com/joshuacords/liferay-portal/pull/109

i made a small request to update the tests, thanks!

**Comment by BryanEngler** [comment]
ci:stop

**Pull request closed without merging**